probes: remove HAVE_FIB_LOOKUP leftovers

rgo3 opened the pull request:
As a follow-up to https://github.com/cilium/cilium/pull/29348, this PR removes leftovers related to HAVE_FIB_LOOKUP.

rgo3 committed:
probes: rename header generation test input string

This commit renames the test input for the feature header generation
test from HAVE_FIB_LOOKUP to HAVE_FEATURE_MACRO.

This is in prepation of a following commit where we remove the probing
for HAVE_FIB_LOOKUP as it is no longer needed. HAVE_FEATURE_MACRO is
test-specific and will not show up when grepping for other feature
macros that are in use in the code base.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>

julianwiedmann commented on bpf/include/bpf/helpers_xdp.h:
```diff
@@ -8,7 +8,7 @@
 #include "compiler.h"
 #include "helpers.h"
-#include "features_xdp.h"
+#include "features.h"
```
https://github.com/cilium/cilium/pull/29377 says hi :grin:

rgo3 replied:
Ah I knew this was too aggressive :grin:. I'll drop some stuff from this PR then.

rgo3 committed:
probes: remove HAVE_FIB_LOOKUP leftovers

As a follow-up to #29348, this commit removes probes and feature header
generation related to HAVE_FIB_LOOKUP.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>
